src/DotNetWorker.Grpc/GrpcWorker.cs

```diff
@@ -234,6 +240,12 @@ private async Task ProcessRequestCoreAsync(StreamingMessage request)
                     Exception = ex.ToRpcException(),
                     Status = StatusResult.Types.Status.Failure
                 };
+
+                if (ex.InnerException is TaskCanceledException)
+                {
+                    Console.WriteLine("TaskCanceledException withing handle invocation request");
+                    response.Result.Status = StatusResult.Types.Status.Cancelled;
```

Comment:
How is host handling this Cancelled status? I wonder we should just send Failure as the status here. What if customer function has some code which is making an http call using HttpClient/ Some Azure SDK to talk to Azure resources and a TaskCanceledException was thrown from that. The function failed in that case. Should we not report that as a failure ?

Reply:
discussed offline